bentoml/_internal/io_descriptors/pandas.py

```diff
@@ -485,6 +487,117 @@ def predict(inputs: pd.DataFrame) -> pd.DataFrame: ...
 
         return inst
 
+    async def from_proto(self, request: pb.Request) -> ext.PdDataFrame:
+        """
+        Process incoming protobuf request and convert it to ``pandas.DataFrame``
+
+        Args:
+            request: Incoming RPC request message.
+            context: grpc.ServicerContext
+
+        Returns:
+            a ``pandas.DataFrame`` object. This can then be used
+             inside users defined logics.
+        """
+        contents = {}
+        if request.HasField("dataframe"):
```

Comment:
Hm, are we not supporting `ndarray` input?

Reply:
nope.